src/composer.ts

```diff
@@ -32,8 +35,9 @@ export default abstract class implements Composable {
    * @throws NO_ADDRESSES_PROVIDED if the network is not RSK Mainnet or RSK Testnet and the options parameter was not provided in the constructor - KB005.
    */
   private async _detectNetwork() {
+    const networkId = await this.web3.eth.net.getId();
```

Comment:
I think we should do the following:
- If network ID is provided via constructor, use that network ID
- Otherwise, get network ID and
  - If network ID is obtained, use that network ID
  - Otherwise throw error